src/sparse/impl/KokkosSparse_spiluk_numeric_impl.hpp

```diff
@@ -0,0 +1,509 @@
+      auto col          = U_entries(k);
+      U_values(k)       = 0.0;
+      iw(my_league,col) = k;
+    });
+
+    team.team_barrier();
+	
+    //Unpack the ith row of A
+    k1 = A_row_map(rowid);
+    k2 = A_row_map(rowid+1);
+    Kokkos::parallel_for( Kokkos::TeamThreadRange( team, k1, k2 ), [&] ( const size_type k ) {
+      auto col  = A_entries(k);
+      auto ipos = iw(my_league,col);
+      if (col < rowid)
+        L_values(ipos) = A_values(k);
+      else
+        U_values(ipos) = A_values(k);	  
+    });
+
+    team.team_barrier();
+	
+    //Eliminate prev rows
+    k1 = L_row_map(rowid); 
+    k2 = L_row_map(rowid+1);
+#ifdef KEEP_DIAG
+    for (auto k = k1; k < k2-1; ++k) {
+#else
+    for (auto k = k1; k < k2; ++k) {
+#endif
+      auto prev_row = L_entries(k);
+#ifdef KEEP_DIAG
+      auto fact = L_values(k) / U_values(U_row_map(prev_row));
+#else
+      auto fact = L_values(k) * U_values(U_row_map(prev_row));
+#endif
+      if ( my_team == 0 ) L_values(k) = fact; 
+
+      team.team_barrier();
+
+      Kokkos::parallel_for( Kokkos::TeamThreadRange( team, U_row_map(prev_row)+1, U_row_map(prev_row+1) ), [&] ( const size_type kk ) {
+        auto col  = U_entries(kk);
+        auto ipos = iw(my_league,col);
+        if (ipos != -1) {
+          auto lxu = -U_values(kk) * fact;
+          if (col < rowid)
+            L_values(ipos) += lxu;
+          else
+            U_values(ipos) += lxu;
+        }
+      });// end for kk
+
+      team.team_barrier();
+    }// end for k
+
+    if ( my_team == 0 ) {
+#ifdef KEEP_DIAG
+      if (U_values(iw(my_league,rowid)) == 0.0) {
+        U_values(iw(my_league,rowid)) = 1e6;
+      }
+#else
+      if (U_values(iw(my_league,rowid)) == 0.0) {
+        U_values(iw(my_league,rowid)) = 1e6;
+      }
+      else {
+        U_values(iw(my_league,rowid)) = 1.0 / U_values(iw(my_league,rowid));
+      }
+#endif
+    }
+
+    team.team_barrier();
+
+    //Reset
+    k1 = L_row_map(rowid); 
+    k2 = L_row_map(rowid+1);
+#ifdef KEEP_DIAG
+    Kokkos::parallel_for( Kokkos::TeamThreadRange( team, k1, k2-1 ), [&] ( const size_type k ) {
+      iw(my_league,L_entries(k)) = -1;
+    });
+#else
+    Kokkos::parallel_for( Kokkos::TeamThreadRange( team, k1, k2 ), [&] ( const size_type k ) {
+      iw(my_league,L_entries(k)) = -1;
+    });
+#endif
+
+    k1 = U_row_map(rowid); 
+    k2 = U_row_map(rowid+1);
+    Kokkos::parallel_for( Kokkos::TeamThreadRange( team, k1, k2 ), [&] ( const size_type k ) {
+      iw(my_league,U_entries(k)) = -1;
+    });
+  }
+};
+
+template <class IlukHandle,
+          class ARowMapType,
+          class AEntriesType,
+          class AValuesType,
+          class LRowMapType,
+          class LEntriesType,
+          class LValuesType,
+          class URowMapType,
+          class UEntriesType,
+          class UValuesType>
+void iluk_numeric ( IlukHandle& thandle,
+                    const ARowMapType&  A_row_map,
```

Comment:
Might need a refactor, not just here in other places as well, to use the Crsmatrix instead of the three pointers .